### PR TITLE
chore: update `pyproject.toml`

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
Changes proposed by this PR can be summarized as follows:

* Move LICENSE file inclusion to `setup.py` by adding a `license_files` argument to `setup()`. [Source](https://stackoverflow.com/a/66443941/14081966), thereby eliminating the purpose of `MANIFEST.in`
* Move `isort` and `pytest` configuration to `pyproject.toml`. 

Follow up suggestions, `pyproject.toml` is the new standard for python packaging, would recommend moving all packaging related metadata to `pyproject.toml`. It eliminates the need for updating version's explicitly using the `flit` backend instead of `setuptools` and also allows for placing all metadata within one single file. (for the sad exception of `flake8` which doesn't support `pyproject.toml` as of now)

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
@pacman100

**Comment by pacman100** [comment]
Hello, to be inline with other projects in 🤗 ecosystem, gently pinging @sgugger for his review on this.

**Comment by sgugger** [comment]
You should move to ruff for isort/flake8 as it's much faster. A sample config can be found [here](https://github.com/huggingface/transformers/blob/main/setup.cfg).

**Comment by SauravMaheshkar** [comment]
Thanks for the suggestion, addressed in [9990189](https://github.com/huggingface/peft/pull/125/commits/99901896ccea794fd012b81fbf747c3476d98388)

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
 target-version = ['py36']
+
+[tool.ruff]
+ignore = ["C901", "E501", "E741", "W605"]
+select = ["C", "E", "F", "I", "W"]
+line-length = 119
+
+[tool.ruff.isort]
+default_section = "FIRSTPARTY"
+known_first_party = "pet"
```
Comment (pacman100) [inline pyproject.toml:12]:
```suggestion
known_first_party = "peft"
```

**Review by pacman100 — COMMENTED** [review]
Hello @SauravMaheshkar, Thank you for iterating, you would also need to update the Makefile to use ruff

**Comment by SauravMaheshkar** [comment]
> Hello @SauravMaheshkar, Thank you for iterating, you would also need to update the Makefile to use ruff

Addressed in [94f00b7](https://github.com/huggingface/peft/pull/125/commits/94f00b7d272932ba8f16e6b3ec485eefe900bc70)

**Review by pacman100 — APPROVED** [review]
Thank you @SauravMaheshkar, LGTM! 🤗

**Pull request merged**